cunumeric/module.py

```diff
@@ -186,7 +201,9 @@ def identity(n, dtype=float):
     return eye(N=n, M=n, dtype=dtype)
 
 
-def ones(shape, dtype=np.float64) -> ndarray:
+def ones(
+    shape: Union[int, tuple[int, ...]], dtype: npt.DTypeLike = np.float64
```

Comment:
I guess I would propose something like 
```
NdShape = tuple[int, ...]
NdShapeLike = int | tuple[int, ...]
```
to specifically denote _Numpy_ shapes (and things that might be converted to them). We definitely need a name that distinguishes from legate's  `Shape`. I'm not sure if renaming `Shape` to something less generic and more descriptive is feasible at this point, but it might be worth considering in order to draw conceptual boundaries more clearly.

Reply:
What's the best order here? Simultaneous PRs to legate and cunumeric? Seems they need to be merged ~simultaneously, and I expect at least one build will have to fail regardless